astrometry/include/PixelMap.h

```diff
@@ -332,5 +334,37 @@ namespace astrometry {
     void operator=(const ColorTerm& rhs) = delete;
   
   };
+
+  class ASTMap: public PixelMap {
+    // Class representing a constant shift of position
+  public:
+    ASTMap(const ast::Mapping& mapping_, std::string name_=""):
+      PixelMap(name_), mapping(mapping_) {}
```

Comment:
I think it would be much better to make the `mapping` data member a `std::shared_ptr<ast::Mapping>` and call `copy()` here.  You don't actually care about being able to share pointers, and all else being equal I'd prefer `unique_ptr`, but once you have a `shared_ptr` (and that's what methods like `Mapping::copy` return) you're stuck with it.  The important thing is that the `mapping_` could be an instance of a derived class of `Mapping`, and by copy-constructing it you're slicing off any extra stuff the derived class might have added.

Normally that would result in really horrible bugs, and the copy constructor would have been disabled to protect against that, and I find it strange that that's not the case in `Mapping`.  I'm sure it's related to the fact that the real inheritance is happening in AST C code and the C++ inheritance is just a facade around that, but I still think it's much safer to hold by pointer and avoid the type-slicing yourself.

Reply:
I think I've implemented what you suggest. Can you check and see if I did what you meant? I also changed `duplicate` so that it no longer does a copy.